pkg/ipam/address/address.go

```diff
@@ -187,11 +189,15 @@ func (a api) Delete(ctx context.Context, id string) error {
 	if err != nil {
 		return fmt.Errorf("could not execute address delete request: %w", err)
 	}
+	defer func() {
+		err = httpResponse.Body.Close()
+	}()
+
 	if httpResponse.StatusCode >= 500 && httpResponse.StatusCode < 600 {
 		return fmt.Errorf("could not execute address delete request, got response %s", httpResponse.Status)
 	}
 
-	return httpResponse.Body.Close()
+	return nil
```

Comment:
`err` is initialized to `nil`, making the explicit `nil` redundant - but no idea if having it explicit is better or not.

Reply:
I think I'd prefer not having it explicit, in case someone changes the function in the future, adding a branch where `err` is set but not returning right away for some reason and the return at the end throws away the error because of this